src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/processors/VariantToVariantAnnotationProcessor.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * See 'VCF output' in http://www.ensembl.org/info/docs/tools/vep/vep_formats.html
+ *
+ * Multiple VEP consequences are separated by ','. Data fields are encoded separated by "|";
+ * the order of fields is written in the VCF header. Unpopulated fields are represented by an empty string.
+ * Consequence field can contain multiple values.
+ */
+public class VariantToVariantAnnotationProcessor implements ItemProcessor<Variant, VariantAnnotation> {
+    protected static Logger logger = LoggerFactory.getLogger(VariantToVariantAnnotationProcessor.class);
+
+    @Override
+    public VariantAnnotation process(Variant variant) {
+        logger.trace("Converting variant {} into VariantAnnotation", variant);
+
+        Map<String, VariantSourceEntry> sourceEntries = variant.getSourceEntries();
+
+        List<Csq> csqs = new ArrayList<>();
+
+        for (VariantSourceEntry sourceEntry : sourceEntries.values()) {
+            String csqValue = sourceEntry.getAttribute("CSQ");
+            csqs.addAll(parseCsqValue(csqValue));
+        }
+
+        if (!csqs.isEmpty()) {
+            VariantAnnotation variantAnnotation = new VariantAnnotation(variant.getChromosome(), variant.getStart(),
+                                                                        variant.getEnd(), variant.getReference(),
+                                                                        variant.getAlternate());
+
+            variantAnnotation.setConsequenceTypes(extractConsequenceTypesFromCsq(csqs));
+            variantAnnotation.setHgvs(extractHgvsFromCsq(csqs));
+
+            return variantAnnotation;
+        }
+
+        return null;
+    }
+
+    private List<ConsequenceType> extractConsequenceTypesFromCsq(List<Csq> csqs) {
+        return csqs.stream().map(this::csqToConsequenceTypeMapper).collect(Collectors.toList());
+    }
+
+    private List<String> extractHgvsFromCsq(List<Csq> csqs) {
+        return csqs.stream().map(csq -> Arrays.asList(csq.getHgvsC(), csq.getHgvsP())).flatMap(Collection::stream)
+                .filter(StringUtils::isNotBlank).collect(Collectors.toList());
+    }
+
+    private ConsequenceType csqToConsequenceTypeMapper(Csq csq) {
+        ConsequenceType consequenceType = new ConsequenceType();
+        consequenceType.setcDnaPosition(csq.getcDNAposition());
+        consequenceType.setCdsPosition(csq.getCdsPosition());
+        consequenceType.setBiotype(csq.getBiotype());
+        consequenceType.setEnsemblGeneId(csq.getGene());
+        consequenceType.setEnsemblTranscriptId(csq.getFeature());
+        consequenceType.setStrand(csq.getStrand());
+        consequenceType.setCodon(csq.getCodons());
+        consequenceType.setSoTermsFromSoNames(Arrays.asList(csq.getConsequence().split(",")));
```

Comment:
I feel we are creating many layers by forcing the dependency on Biodata. Given that we already have our own implementation of the VEP output reader and `VariantAnnotation` writer, I would suggest to expend some extra time on this and build a model that really suits us.